grpc/src/main/java/com/linecorp/armeria/server/grpc/GrpcHealthCheckService.java

```diff
@@ -0,0 +1,314 @@
+            Consumer<String> healthCheckUpdateListener,
+            Collection<ListenableHealthChecker> listenableHealthCheckers) {
+        listenableHealthCheckers.forEach(
+                lhc -> lhc.addListener(healthChecker -> healthCheckUpdateListener.accept("")));
+    }
+
+    private static void setHealthUpdateListenerToGrpcServiceHealthCheckers(
+            Consumer<String> healthCheckUpdateListener,
+            Iterable<Entry<String, ListenableHealthChecker>> listenableHealthCheckers) {
+        listenableHealthCheckers.forEach(entry -> entry.getValue().addListener(
+                healthChecker -> healthCheckUpdateListener.accept(entry.getKey())));
+    }
+
+    private final SettableHealthChecker serverHealth;
+    private final Set<ListenableHealthChecker> serverHealthCheckers;
+    private final Map<String, ListenableHealthChecker> grpcServiceHealthCheckers;
+    private final Multimap<String, Entry<StreamObserver<HealthCheckResponse>, ServingStatus>> watchers =
+            HashMultimap.create();
+    @Nullable
+    private Server server;
+
+    GrpcHealthCheckService(
+            Set<ListenableHealthChecker> serverHealthCheckers,
+            Map<String, ListenableHealthChecker> grpcServiceHealthCheckers,
+            List<HealthCheckUpdateListener> updateListeners
+    ) {
+        serverHealth = new SettableHealthChecker(false);
+        if (!updateListeners.isEmpty()) {
+            addServerHealthUpdateListener(updateListeners);
+        }
+        this.serverHealthCheckers = ImmutableSet.<ListenableHealthChecker>builder()
+                                                .add(serverHealth)
+                                                .addAll(serverHealthCheckers)
+                                                .build();
+        this.grpcServiceHealthCheckers = grpcServiceHealthCheckers;
+        final Consumer<String> healthCheckUpdateListener = provideInternalHealthUpdateListener();
+        setHealthUpdateListenerToServerHealthCheckers(healthCheckUpdateListener, this.serverHealthCheckers);
+        setHealthUpdateListenerToGrpcServiceHealthCheckers(healthCheckUpdateListener,
+                                                           this.grpcServiceHealthCheckers.entrySet());
+    }
+
+    @Override
+    public void check(HealthCheckRequest request, StreamObserver<HealthCheckResponse> responseObserver) {
+        final String service = request.getService();
+        final ServingStatus status = checkServingStatus(service);
+        if (status == ServingStatus.SERVICE_UNKNOWN) {
+            responseObserver.onError(Status.NOT_FOUND
+                                             .withDescription(String.format(
+                                                     "The service name(%s) is not registered in this service",
+                                                     service))
+                                             .asRuntimeException());
```

Comment:
Indentation:
```suggestion
            responseObserver.onError(Status.NOT_FOUND
                                           .withDescription(String.format(
                                                   "The service name (%s) is not registered in this service",
                                                   service))
                                           .asRuntimeException());
```

Reply:
I think it's better to leave this part as it is according to the formatted results
ref. https://github.com/line/armeria/pull/3963#discussion_r764511674